x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/SequenceMatcher.java

```diff
@@ -253,19 +253,15 @@ void until(Iterable<KeyAndOrdinal> markers) {
      * This allows the matcher to keep only the last match per stage
      * and adjust insertion positions.
      */
-    void trim(boolean everything) {
+    void trim(Ordinal ordinal) {
```

Comment:
Wondering if `@Nullable` would be welcome here, as an indicator of `null` as a branching value?

Reply:
I'm not a fan of `@Nullable` since it doesn't enforce anything nor it is used consistently through-out the code.